protostuff-json/src/main/java/io/protostuff/JsonInput.java

```diff
@@ -197,12 +196,12 @@ public JsonInput reset()
                         // skip the scalar elements
                         while (parser.nextToken() != END_ARRAY)
                             ;
-
-                        continue;
                     }
-
-                    throw new JsonInputException("Unknown field: " + name + " on message " +
-                            schema.messageFullName());
+					else

```

Comment:
Please reformat code - change tabs to spaces. Formatter settings: https://github.com/protostuff/protostuff/blob/master/styleGuidelines.xml (eclipse)